service/matching/db_task_manager.go

```diff
@@ -92,26 +92,39 @@ type (
 		taskReader                dbTaskReader
 		taskWriter                dbTaskWriter
 		maxDeletedTaskIDInclusive int64 // in mem only
+		group                     goro.Group
+		dbTaskDeletionInterval    time.Duration
+		dbTaskUpdateAckInterval   time.Duration
+		dbTaskUpdateQueueInterval time.Duration
```

Comment:
why do you need to use goro here?

Reply:
When the Stop() is called on db task manager, and the shutdownDownChan is closed, we need to wait for ongoing write to finish, don't we? That's why I added it. Also, we need to be able to cancel ongoing read immediately. Using the goro to do cancel is simpler than WithCancel.